pandas/core/ops/__init__.py

```diff
@@ -322,11 +318,14 @@ def dispatch_to_series(left, right, func, str_rep=None, axis=None):
         bm = left._mgr.apply(array_op, right=right)
         return type(left)(bm)
 
-    elif isinstance(right, ABCDataFrame):
+    array_op = get_array_op(func, str_rep=str_rep)
```

Comment:
might as well do this once up top instead of here and on L317

Reply:
Ah, missed the existing one, thanks